scripts/imgtool/keys.py

```diff
@@ -0,0 +1,129 @@
+"""
+Cryptographic key management for imgtool.
+"""
+
+from __future__ import print_function
+
+from Crypto.Hash import SHA256
+from Crypto.PublicKey import RSA
+from Crypto.Signature import PKCS1_v1_5
+from ecdsa import SigningKey, NIST256p, util
+from pyasn1.type import namedtype, univ
+from pyasn1.codec.der.encoder import encode
+
+class RSAPublicKey(univ.Sequence):
+    componentType = namedtype.NamedTypes(
+            namedtype.NamedType('modulus', univ.Integer()),
+            namedtype.NamedType('publicExponent', univ.Integer()))
+
+class RSA2048():
```

Comment:
I thought about the base class, but wasn't sure what advantage it gives? Maybe documentation, but I'm not aware of python actually enforcing anything related to typing, even at runtime.